File: a_python/c_oop/d_encapsulation/b_exercise/c_football_team_generator/project/team.py
from project.player import Player
class Team:
    def __init__(self, name, rating):
        self.__name = name
        self.__rating = rating
        self.__players = []

    def add_player(self, player):
        if player not in self.__players:
            self.__players.append(player)
            return f'Player {player.name} joined team {self.__name}'
        else:
            return f'Player {player.name} has already joined'

    def remove_player(self, player_name):
        for p in self.__players:
            if p.name == player_name:
                self.__players.remove(p)
                return p
        return f'Player {player_name} not found'

File: 03_oop/01_encapsulation/Examples/00_People/Team.cs
namespace People;

public class Team
{
    private List<Person> _firstTeam;
    private List<Person> _reserveTeam;

    public Team(string name)
    {
        Name = name;
        _firstTeam = new List<Person>();
        _reserveTeam = new List<Person>();
    }

    public string Name { get; private set; }

    public IReadOnlyCollection<Person> FirstTeam { get => _firstTeam; }

    public IReadOnlyCollection<Person> ReserveTeam { get => _reserveTeam; }

    public void AddPlayer(Person player)
    {
        if (player.Age < 40) _firstTeam.Add(player);
        else _reserveTeam.Add(player);
    }
}